go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -130,66 +130,64 @@ func (c *addressManager) Run(stopChan <-chan struct{}, doneWg *sync.WaitGroup) {
 		return true, addrChan, nil
 	}
 	c.addHandlerForPrimaryAddrChange()
-	c.runInternal(stopChan, doneWg, subscribe)
+	doneWg.Add(1)
+	go func() {
+		c.runInternal(stopChan, subscribe)
+		doneWg.Done()
+	}()
 }
 
 // runInternal can be used by testcases to provide a fake subscription function
 // rather than using netlink
-func (c *addressManager) runInternal(stopChan <-chan struct{}, doneWg *sync.WaitGroup, subscribe subscribeFn) {
-	doneWg.Add(1)
-	go func() {
-		defer doneWg.Done()
-
-		addressSyncTimer := time.NewTicker(30 * time.Second)
-		defer addressSyncTimer.Stop()
+func (c *addressManager) runInternal(stopChan <-chan struct{}, subscribe subscribeFn) {
+	addressSyncTimer := time.NewTicker(30 * time.Second)
+	defer addressSyncTimer.Stop()
 
-		subscribed, addrChan, err := subscribe()
-		if err != nil {
-			klog.Errorf("Error during netlink subscribe for IP Manager: %v", err)
-		}
-
-		for {
-			select {
-			case a, ok := <-addrChan:
-				addressSyncTimer.Reset(30 * time.Second)
-				if !ok {
-					if subscribed, addrChan, err = subscribe(); err != nil {
-						klog.Errorf("Error during netlink re-subscribe due to channel closing for IP Manager: %v", err)
-					}
-					continue
-				}
-				addrChanged := false
-				if a.NewAddr {
-					addrChanged = c.addAddr(a.LinkAddress)
-				} else {
-					addrChanged = c.delAddr(a.LinkAddress)
+	subscribed, addrChan, err := subscribe()
+	if err != nil {
+		klog.Errorf("Error during netlink subscribe for IP Manager: %v", err)
```

Comment:
nit: this is not netlink subscribe right? copy/paste snafu smell..
~and secondly.. if we fail to subscribe shouldn't we return and retry or something? I find it weird that we are continuing to process stuff in the addrChan by simply logging the error...~ nvm we are checking this again in the select block and returning there..